src/app/components/project-map/project-map.component.ts

```diff
@@ -1086,6 +1096,71 @@ export class ProjectMapComponent implements OnInit, OnDestroy {
     instance.project = this.project;
   }
 
+  private async getHttpControllerNotifications(): Promise<void> {
+    let url = this.notificationService.getPathControllerNotifications(this.controller);
+    var gns3Headers = new Headers();
+    gns3Headers.append('Authorization', `Bearer ${this.controller.authToken}`);
+    var requestOptions = {
+      headers: gns3Headers,
+    };
+    const response = await fetch(url, requestOptions);
+    if (response.body == null) {
+      throw new Error('Http Notifications Failed');
+    }
+    const reader = response.body.getReader();
+    const cd = this.cd;
+    const controllerHttpStream = async () => {
+      const { done, value } = await reader.read();
+      const textDecoder = new TextDecoder();
+      if (done) {
+        return;
+      }
+      const notificationValue = textDecoder.decode(value);
+      const validStreamJson = notificationValue
+        .replace('[{', '{')
+        .replace('},', '}')
+        .replace(',{', '{')
+        .replace('}]', '}');
+      let data = JSON.stringify(validStreamJson);
+      this.projectWebServiceHandler.handleMessage(JSON.parse(data));
+      cd.markForCheck();
```

Comment:
This could a separate method/function since there is the same code below for the project HTTP stream.